### PR TITLE
Exclusion Report 

### PR DESCRIPTION
**Pull request by amichard**
#1399 #1400 #1401 

Added save functionality for the exclusion report

Changelog:
- Added Backend models for Exclusion Report
- Added Serializers to save the records
- Added a check in the viewset so it knows which serializer it should use
- Added toastr alert for saving an exclusion report draft
- Clicking on an exclusion report should show the proper details
- Fixed GH Genius not filtering by class

### CODE REVIEW

**Comment by amichard** [comment]
[Exclusion Report | Save a draft |5|](https://trello.com/c/ZslHs17z/1563-exclusion-report-save-a-draft-5)

**Comment by amichard** [comment]
[System Alerts | Save a Draft Exclusion Report |1|](https://trello.com/c/J6dcmwLj/1568-system-alerts-save-a-draft-exclusion-report-1)

**Comment by amichard** [comment]
[Exclusion Report | Edit a draft |3|](https://trello.com/c/D6knfdFA/1564-exclusion-report-edit-a-draft-3)

File: backend/api/serializers/ComplianceReport.py [hunk 1]
```diff
@@ -451,6 +446,10 @@ def update(self, instance, validated_data):
 
         if instance.read_only and not self.disregard_status:
             raise PermissionDenied('Cannot modify this compliance report')
+        
+        # if 'agreement' in validated_data:
```
Comment (plasticviking) [inline backend/api/serializers/ComplianceReport.py]: This could get removed?

File: backend/api/serializers/ExclusionReport.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        model = ExclusionAgreement
+        fields = ('records',)
+
+
+class ExclusionReportDetailSerializer(serializers.ModelSerializer):
+    """
+    Detail Serializer for the Exclusion Report
+    """
+    status = ComplianceReportStatusSerializer(read_only=True)
+    type = ComplianceReportTypeSerializer(read_only=True)
```
Comment (plasticviking) [inline backend/api/serializers/ExclusionReport.py]: My IDE thinks type is a reserved word. It seems to work but maybe rename it for clarity.

File: backend/api/serializers/ComplianceReport.py (hunk 1 above)
Comment (amichard) [reply backend/api/serializers/ComplianceReport.py]: good catch

File: backend/api/serializers/ExclusionReport.py (hunk 2 above)
Comment (amichard) [reply backend/api/serializers/ExclusionReport.py]: ill take a look at this. i prefer to keep the name similar to the one in the model to lessen confusion

Comment (amichard) [reply backend/api/serializers/ExclusionReport.py]: Ya is there a way for the IDE to ignore this instead? type is quite spread around our Compliance Report serializers and we're into this too deep to make changes.

**Comment by amichard** [comment]
I'll add unit tests for this later

**Comment by kuanfandevops** [comment]
@amichard the migration number conflicts with existing one, need to be renamed

**Pull request merged**